components/Button.tsx

```diff
@@ -0,0 +1,57 @@
+import Image from 'next/image';
+import { MouseEventHandler } from 'react';
+
+interface ButtonProps {
+  type: 'button' | 'submit';
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  iconUrl?: string;
+  iconAlt?: string;
+  content?: string;
+  design: 'solid' | 'outlined';
+  disabled: boolean;
+  className?: string;
+}
+
+const Button = ({
+  type = 'button',
+  onClick,
+  iconUrl,
+  iconAlt = '',
+  content = '',
+  design = 'solid',
+  disabled = false,
+  className = '',
+}: ButtonProps) => {
+  let buttonStyle =
+    'flex flex-row items-center justify-center gap-1 ' +
+    'w-full h-[58px] md:h-[72px] rounded-lg ' +
+    'text-lg md:text-xl font-semibold ';
+
+  if (design === 'solid' && !disabled)
+    buttonStyle +=
+      'bg-orange-300 hover:bg-orange-200 focus:bg-orange-200 text-gray-50 ';
+  if (design === 'solid' && disabled)
+    buttonStyle += 'bg-gray-100 text-gray-50 ';
+  if (design === 'outlined' && !disabled)
+    buttonStyle +=
+      'border border-orange-300 hover:border-orange-200 focus:border-orange-200 ' +
+      'text-orange-300 hover:text-orange-200 focus:text-orange-200 ';
+  if (design === 'outlined' && disabled)
+    buttonStyle += 'border border-gray-100 text-gray-100 ';
```

Comment:
💊 취향차이이긴 한데 아래 처럼하면 조금 더 조건문이 눈에 잘 들어올것같아요!
```suggestion
switch (design) {
  case 'solid':
    buttonStyle += disabled
      ? 'bg-gray-100 text-gray-50 '
      : 'bg-orange-300 hover:bg-orange-200 focus:bg-orange-200 text-gray-50 ';
    break;

  case 'outlined':
    buttonStyle += disabled
      ? 'border border-gray-100 text-gray-100 '
      : 'border border-orange-300 hover:border-orange-200 focus:border-orange-200 ' +
        'text-orange-300 hover:text-orange-200 focus:text-orange-200 ';
    break;
}
```